### PR TITLE
fix(platform): table p13 selection a11y fixes

### LINKED ISSUE #10868: [P3.1][Sourcing][Accessibility][p13n dialog] Checkbox is not being read by screenreader while browsing the 'Columns' inside <fdp-table>'s p13n dialog
#### Is this a bug, enhancement, or feature request?
Bug

#### Describe your proposal.
The checkbox and its status should be announced correctly to the screenreader when accessed.

#### Can you handle that on the application side
No

#### Which versions of Angular and Fundamental Library for Angular are affected? Please, specify the exact version. (If this is a feature request, use current version.)
- Angular 15
- F-NGX Platform: 0.43.14

#### If this is a bug, please provide steps for reproducing it; the exact components you are using;
Please see the `Personalization Dialog` section under this example:
https://ng-15-downport--fundamental-ngx.netlify.app/#/platform/table/p13-dialog-table

Screenshot 2023-11-01 at 5 09 32 PM

#### In case this is Accessibility related topic, did you consult with an accessibility expert? If not, please do so and share their recommendations. 
UI type, information, properties of all UI elements should be conveyed correctly once to the screen reader users when accessed.

### PR DESCRIPTION
**Pull request by mikerodonnell89**
fixes #10868

### CODE REVIEW

**Review by g-cheishvili — APPROVED** [review]

File: libs/platform/table/components/table-p13-dialog/columns/columns.component.html [hunk 1]
```diff
@@ -75,8 +75,8 @@ <h4 fd-title [headerSize]="4">
                 (click)="_moveActiveToBottom()"
             ></button>
         </fd-toolbar>
-        <ul fd-list>
-            <li fd-list-item>
+        <ul fd-list [selection]="true" aria-multiselectable="true">
```
Comment (InnaAtanasova) [inline libs/platform/table/components/table-p13-dialog/columns/columns.component.html:78]: `aria-multiselectable` is used in roles grid, listbox, tablist and tree. Your `ul` element should have one of these roles. Additionally, the list elements should be with role `option` and `aria-selected` set to true or false.

**Review by InnaAtanasova — CHANGES_REQUESTED** [review]

File: libs/platform/table/components/table-p13-dialog/columns/columns.component.html (hunk 1 above)
Comment (InnaAtanasova) [reply libs/platform/table/components/table-p13-dialog/columns/columns.component.html:78]: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-multiselectable

Comment (InnaAtanasova) [reply libs/platform/table/components/table-p13-dialog/columns/columns.component.html:78]: https://www.digitala11y.com/aria-multiselectable-properties/

Comment (mikerodonnell89) [reply libs/platform/table/components/table-p13-dialog/columns/columns.component.html:78]: @InnaAtanasova the correct roles and aria-selected are being applied to the elements via the directives. If you inspect the html:
Screenshot 2024-01-02 at 5 51 54 PM

**Review by InnaAtanasova — APPROVED** [review]

File: libs/platform/table/components/table-p13-dialog/columns/columns.component.html (hunk 1 above)
Comment (InnaAtanasova) [reply libs/platform/table/components/table-p13-dialog/columns/columns.component.html:78]: maybe we should make `multiselectable` a prop of the directive in this case. Could be in a separate PR. I will approve this one.

**Pull request merged**